src/slider/tests/functional/Slider.ts

```diff
@@ -0,0 +1,250 @@
+		'each component of a slider should be visible'(this: any) {
+			return getPage(this)
+				.findByCssSelector(`.s2.${css.root}`)
+				.isDisplayed()
+				.findByCssSelector(`.${css.input}`)
+				.isDisplayed()
+				.end()
+				.findByCssSelector(`.${css.track}`)
+				.isDisplayed()
+				.end()
+				.findByCssSelector(`.${css.output}`)
+				.isDisplayed()
+				.end()
+
+				.getSize()
+				.then(({ height, width }: { height: number; width: number; }) => {
+					assert.isAbove(height, 0, 'The height of the slider should be greater than zero.');
+					assert.isAbove(width, 0, 'The width of the slider should be greater than zero.');
+				})
+
+				.end();
+		},
+		'label should be as defined'(this: any) {
+			return getPage(this)
+				.findByCssSelector(`.s3.${css.root}`)
+				.getVisibleText()
+				.then((text: string) => {
+					assert.include(text, 'Vertical Slider with default properties.');
+				})
+				.end();
+		},
+		'slider value should be consistent in different part of the UI'(this: any) {
+			let command = getPage(this)
+				.findByCssSelector(`.s2.${css.root}`);
+			return checkValue(command)
+				.end();
+		},
+		'slider should be functional with mouse'(this: any) {
+			const { browserName, mouseEnabled } = this.remote.environmentType;
+			if (!mouseEnabled) {
+				this.skip('Test requires mouse interactions.');
+			}
+			if (browserName.toLowerCase() === 'internet explorer') {
+				this.skip('mouse movements doesn\'t work in IE.');
+			}
+
+			let sliderValues: number[] = [];
+			let command = getPage(this)
+				.findByCssSelector(`.s3.${css.root}`);
+			command = checkValue(command, sliderValues);
+
+			command = slide(command, 1, -30);
+			command = checkValue(command, sliderValues);
+
+			command = slide(command, 1, -40);
+			command = checkValue(command, sliderValues);
+
+			return command
+				.then(() => {
+					assert.lengthOf(sliderValues, 3);
+					assert.isTrue(sliderValues[1] > sliderValues[0] && sliderValues[2] > sliderValues[1]);
+				})
+				.end();
+		},
+		'slider should be functional with up and down arrow keys'(this: any) {
+			const { browserName, supportsKeysCommand } = this.remote.environmentType;
+			if (!supportsKeysCommand) {
+				this.skip('Arrow keys required for tests.');
+			}
+			if (browserName.toLowerCase() === 'safari' || browserName.toLowerCase() === 'internet explorer') {
+				this.skip('pressKeys with arrow keys doesn\'t work in iphone and IE.');
+			}
+
+			let sliderValues: number[] = [];
+			let command = getPage(this)
+				.findByCssSelector(`.s3.${css.root}`);
+			command = checkValue(command, sliderValues)
+				.click()
+				.pressKeys([keys.ARROW_UP, keys.ARROW_UP]);
+
+			command = checkValue(command, sliderValues)
+				.pressKeys(keys.ARROW_DOWN);
+
+			return checkValue(command, sliderValues)
+				.then(() => {
+					assert.lengthOf(sliderValues, 3);
+					assert.isTrue(sliderValues[1] > sliderValues[2] && sliderValues[2] > sliderValues[0]);
+				})
+				.end();
+		}
+	}
```

Comment:
One other potential test: clicking on the UI elements (or at least on the x,y position of the visible UI) sets focus on the `<input>`

Reply:
Added in a bunch of tests for this